### PR TITLE
UV Lamp: The Wrenchening

### PR DESCRIPTION
**Pull request by RSG250**
[QoL]
## About the PR 
adds code thats used to secure the high capacity water tanks to floor to the UV lamp 

## Why's this needed? 
Its annoying when other botanists accidentally move the UV lamp out of the perfect position you put it in

## Changelog

```
(u)RSG250:
(+)can now secure the UV lamp to the floor via Wrench or Screwdriver
```

### CODE REVIEW

File: code/obj/machinery/plantpot.dm [hunk 1]
```diff
@@ -1783,6 +1783,18 @@ proc/HYPmutationcheck_sub(var/lowerbound,var/upperbound,var/checkedvariable)
 		else
 			light.disable()
 
+	attackby(obj/item/W as obj, mob/user as mob)
+		if(istool(W, TOOL_SCREWING | TOOL_WRENCHING))
```
Comment (Sovexe) [inline code/obj/machinery/plantpot.dm:1787]: Use the macros

```dm
#define isscrewingtool(x) (istool(x, TOOL_SCREWING) || (istype(x, /obj/item/reagent_containers) && x:reagents:has_reagent("screwdriver")) ) //the joke is too good
#define iswrenchingtool(x) (istool(x, TOOL_WRENCHING))
```

**Review by Sovexe — CHANGES_REQUESTED** [review]

File: code/obj/machinery/plantpot.dm [hunk 2]
```diff
@@ -1783,6 +1783,28 @@ proc/HYPmutationcheck_sub(var/lowerbound,var/upperbound,var/checkedvariable)
...
+			if(!src.anchored)
+				user.visible_message("<b>[user]</b> secures the [src] to the floor!")
+				playsound(src.loc, "sound/items/Screwdriver.ogg", 100, 1)
+				src.anchored = 1
+			else
+				user.visible_message("<b>[user]</b> unbolts the [src] from the floor!")
+				playsound(src.loc, "sound/items/Screwdriver.ogg", 100, 1)
+				src.anchored = 0
+			return
+		else if(iswrenchingtool(W))
```
Comment (Sovexe) [inline code/obj/machinery/plantpot.dm:1797]:
```suggestion
		if(isscrewingtool(W) || iswrenchingtool(W))
```

you can still combine them

Comment (RSG250) [reply code/obj/machinery/plantpot.dm:1797]: oh neat

**Review by Sovexe — APPROVED** [review]

**Review by ZeWaka — APPROVED** [review]
Could be more compact but nitpicking

File: code/obj/machinery/plantpot.dm [hunk 3]
```diff
@@ -1783,6 +1783,18 @@ proc/HYPmutationcheck_sub(var/lowerbound,var/upperbound,var/checkedvariable)
...
+		if(isscrewingtool(W) || iswrenchingtool(W))
+			if(!src.anchored)
+				user.visible_message("<b>[user]</b> secures the [src] to the floor!")
+				playsound(src.loc, "sound/items/Screwdriver.ogg", 100, 1)
+				src.anchored = 1
+			else
+				user.visible_message("<b>[user]</b> unbolts the [src] from the floor!")
+				playsound(src.loc, "sound/items/Screwdriver.ogg", 100, 1)
+				src.anchored = 0
+			return
```
Comment (Sovexe) [inline code/obj/machinery/plantpot.dm:1796]:
```suggestion
			if(!src.anchored)
				user.visible_message("<b>[user]</b> secures the [src] to the floor!")
			else
				user.visible_message("<b>[user]</b> unbolts the [src] from the floor!")
			playsound(src.loc, "sound/items/Screwdriver.ogg", 100, 1)
			src.anchored = !src.anchored
```
For Zewaka

**Comment by zjdtmkhzt** [comment]
I was somehow under the impression you could secure those already??
Anyway, good feature!

**Comment by pali6** [comment]
Too late now but we now have a pretty handy action bar for anchoring / unanchoring objects.

**Pull request merged**